lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1633,27 +1650,96 @@ LogicalResult PaddedSharedEncodingAttr::verify(
   if (intervalValues.size() != intervals.size())
     return emitError() << "interval values cannot have duplicates";
 
-  if (order.empty())
-    return emitError() << "order cannot be empty";
+  const auto &ll = linearComponent;
+  // The linear layout should mape from [offset, block] to [dim0..dimN). All
+  // bases should be 0 or power of twos and move in a single direction without
+  // broadcasting
+  for (auto outDim : ll.getInDimNames()) {
+    if (outDim.str() != "offset" && outDim.str() != "block")
+      return emitError() << "linearComponent must have `offset` and `block` as "
+                            "only input dims, found: "
+                         << outDim.str();
+  }
+  if (ll.getNumInDims() != 2)
+    return emitError() << "linearComponent must have `offset` and `block` as "
+                          "only input dims";
```

Comment:
Just check `llvm::equal(ll.getInDimNames(), {str_attr("offset"), str_attr("block")})` or a similar idiom. It is more compact and you also check that the order is as expected.